backend/src/main/java/gov/cdc/usds/simplereport/service/OktaService.java

```diff
@@ -0,0 +1,121 @@
+        userProfileMap.put("login",apiUser.getLoginEmail());
+
+        UserBuilder.instance()
+                .setProfileProperties(userProfileMap)
+                .buildAndCreate(_client);
+    }
+
+    public void updateUser(String oldUsername, ApiUser apiUser) {
+        User user = _client.listUsers(oldUsername, null, null, null, null).single();
+        user.getProfile().setFirstName(apiUser.getNameInfo().getFirstName());
+        user.getProfile().setMiddleName(apiUser.getNameInfo().getMiddleName());
+        user.getProfile().setLastName(apiUser.getNameInfo().getLastName());
+        // Is it our fault we don't accommodate honorific suffix? Or Okta's fault they 
+        // don't have regular suffix? You decide.
+        user.getProfile().setHonorificSuffix(apiUser.getNameInfo().getSuffix());
+        // We assume login == email
+        user.getProfile().setEmail(apiUser.getLoginEmail());
+        user.getProfile().setLogin(apiUser.getLoginEmail());
+        user.update();
+    }
+
+    public void createOrganization(String externalId, String name) {
+        for (OrganizationRole role : OrganizationRole.values()) {
+            GroupBuilder.instance()
+            .setName(generateGroupName(externalId, role))
+            .setDescription(generateGroupDescription(name, role))
+            .buildAndCreate(_client);
+        }
+    }
+
+    public void addUserToOrganization(ApiUser apiUser, String organizationExternalId) {
+        User user = _client.listUsers(apiUser.getLoginEmail(), null, null, null, null).single();
+        String userId = user.getId();
+        // We assume that a user can only be a member of one organization at a time;
+        // if user is in any groups, remove that association first
+        GroupList oldGroups = user.listGroups();
+        oldGroups.forEach(g->{
+            System.out.print(g.toString());
+            if (g.getType() == GroupType.OKTA_GROUP && g.getProfile().getName().endsWith(":USER")) {
+                g.removeUser(userId);
+            }
+        });
+        // Okta SDK's way of adding a user to a group by group name
+        String groupName = generateGroupName(organizationExternalId, OrganizationRole.USER);
+        GroupList groups = _client.listGroups(groupName, null, null);
+        groups.forEach(g->{
+            if (groupName.equals(g.getProfile().getName())) {
+                user.addToGroup(g.getId());
+            }
+        });
+    }
+
+    private String generateGroupName(String externalId, OrganizationRole role) {
+        String suffix = (role == OrganizationRole.ADMIN ) ? ":ADMIN" :
+                        (role == OrganizationRole.USER  ) ? ":USER"  :
+                                                            ":OTHER";
+        return ROLE_PREFIX + externalId + suffix;
+    }
+
+    private String generateGroupDescription(String groupDesc, OrganizationRole role) {
+        String suffix = (role == OrganizationRole.ADMIN ) ? " - Admins" :
```

Comment:
This is an interesting point—should we add a descriptive string to the OrganizationRole enumeration? It seems like the tag and the description should live together, rather than the description living in here.